Userland/Libraries/LibGL/Vertex.cpp

```diff
@@ -295,7 +295,10 @@ void GLContext::gl_vertex(GLfloat x, GLfloat y, GLfloat z, GLfloat w)
         vertex.tex_coords[i] = m_current_vertex_tex_coord[i];
     vertex.normal = m_current_vertex_normal;
 
-    m_vertex_list.append(vertex);
+    // Optimization: by pulling in the Vector size vs. capacity check, we can always perform an unchecked append
+    if (m_vertex_list.size() == m_vertex_list.capacity())
+        m_vertex_list.grow_capacity(m_vertex_list.size() + 1);
+    m_vertex_list.unchecked_append(vertex);
```

Comment:
This one feels weird, as this check should be visible and candidate for inlining in this TU anyway, any more info on this?
Also does this change the growing behavior?

Reply:
This does not affect the growing behavior; `try_append` also grows with `size() + 1`.

I agree that it feels weird, but `Vector::try_append` does not get inlined in a lot of cases. Maybe because there's too much `MUST` and `TRY` going on? I didn't think it would be wise to `ALWAYS_INLINE` the `try_append` method though.